### ISSUE TITLE
Bug: Error when using MeshLambertMaterial

### ISSUE THREAD

**Issue by bennlich** [protectwise/troika#182]
Occasionally, when I create a TroikaThreeText with a MeshLambertMaterial, the text does not render and I see this error:

```
THREE.WebGLProgram: shader error:  1281 35715 false gl.getProgramInfoLog Field names of uniform 'spotLights' differ between VERTEX and FRAGMENT shaders.
```

However, this bug does not occur if the TroikaThreeText is the only object in my scene. It seems to involve some interaction between the TroikaThreeText and other objects in the scene. I will work on narrowing this down, but I'm wondering if anything sticks out to you about this error.

Sincerely loving TroikaThreeText! Thank you!

**Comment by lojjic** [comment]
Sounds like possibly something going wrong with the shader rewriting, but it's not obvious to me at first glance. And I don't know why it would be inconsistent based on other objects in the scene ... very curious.

If you can come up with a minimal test case that would help greatly. If not, maybe pasting the contents of the final shaders from the log would give some clues?

**Comment by lojjic** [comment]
Ah I bet this is due to [these lines](https://github.com/protectwise/troika/blob/9c8c6b1e472f4068a8c6b0f1a6f3e32c935f8077/packages/troika-three-utils/src/DerivedMaterial.js#L361) which rename usages of the `position` uniform in the vertex shader. Since that [`spotLights` struct](https://github.com/mrdoob/three.js/blob/dev/src/renderers/shaders/ShaderChunk/lights_pars_begin.glsl.js#L139) has a `position` property in it, that will also be rewritten, which would obviously cause problems.

This seems like it would fail whenever there are spot lights in the scene, might the "other objects in the scene" trigger happen to be presence of spot lights?

**Comment by bennlich** [comment]
Ahhhh, I see. Thank you. I haven't had time to verify this but it's totally plausible that the "other objects in the scene" do have spotlights. (They are GLBs, so they can contain all manner of things.) I'll try and inspect them soon.

**Comment by lojjic** [comment]
Yep, I can reproduce the bug with MeshLambertMaterial and a SpotLight in the scene, that's definitely the problem. Should be a pretty simple fix.
